docs/DEVELOPMENT.md

```diff
@@ -0,0 +1,402 @@
+  deactivate
+  ```
+
+* Set up VirtualEnv just for this microservice component
+  ```
+  cd components/<component_name>
+
+  python3 -m venv .venv
+  source .venv/bin/activate
+  pip install --upgrade pip
+
+  # install requirements from common
+  pip install -r ../common/requirements.txt
+
+  # install microservice requirements
+  pip install -r requirements.txt
+  ```
+* (For VS Code only) Set up Python interpreter:
+  * Open Command Palette (CMD + SHIFT + P)
+  * Type `Python: Select Interpreter`
+  * Choose your new interpreter
+    * will look something like `./common/.venv/bin/python3`
+
+* (For VS Code Only) Add the following to the `settings.json` in VS Code. The modules should load when you run from command line, and you should see code completion hints.
+  ```json
+  {
+    "terminal.integrated.env.osx": {
+      "PYTHONPATH": "${workspaceFolder}/common/src/"
+    },
+    "python.analysis.extraPaths": [
+      "${workspaceFolder}/common/src/"
+    ]
+  }
+  ```
+
+* You should now be able to load modules and test them locally:
+  ```
+  cd components/<component_name>/src
+  python main.py
+  ```
+
+* Once you finish development for this component, run the following to exit the
+VirtualEnv:
+  ```
+  deactivate
+  ```
+
+###  3.3. <a name='SetupVirtualEnvforcomponentscommon'></a>Set up VirtualEnv for `components/common`
+
+Follow the steps below to create a new VirtualEnv for `components/common` and install required python packages defined in the `requirements.txt`.
+
+* Create a VirtualEnv for a specific component
+```
+cd components/<component_name> # e.g. llm_service
+
+python3 -m venv .venv
+source .venv/bin/activate
+pip install --upgrade pip
+pip install -r requirements.txt
+```
+* (For VSCode only) Set up Python interpreter:
+  * Open Command Palette (CMD + SHIFT + P)
+  * Type `Python: Select Interpreter`
+  * Choose your new interpreter
+    * will look something like `./common/.venv/bin/python3`
+
+* You should now be able to load modules and test them locally:
+  ```
+  # In ./components/<component_name>
+  cd src
+  python
+  ```
+  In REPL (Copy and paste the code)
+  ```python
+  from common.models import User
+  user = User()
+  ```
+
+* Once you finish development for this component, run the following to exit the VirtualEnv:
+  ```
+  deactivate
+  ```
+
+##  4. <a name='DevelopandTestonaGKEcluster'></a>Develop and Test on a GKE cluster
+
+###  4.1. <a name='Doublecheckthecurrentgcloudconfig'></a>Double check the current `gcloud` config
+
+* Please make sure the `gcloud` command set to the current project.
+  ```
+  export PROJECT_ID=<your-dev-project-id>
+  gcloud config set project $PROJECT_ID
+  ```
+
+###  4.2. <a name='SetupWorkloadIdentityUser'></a>Set up Workload Identity User
+
+* Run the following to set up the Kubernetes Service Account (ksa) in your namespace:
+  ```
+  export PROJECT_ID=<your-dev-project-id>
+  export SKAFFOLD_NAMESPACE=$YOUR_GITHUB_ID
+  export GSA_NAME="gke-sa"
+  export KSA_NAME="gke-sa"
```

Comment:
Should this be `ksa`

Reply:
This has to be `gke-sa`, this is defined in the terraform 3-gke stage.